<ui version="4.0" >
 <class>Kleo::Config::SMimeValidationConfigurationWidget</class>
 <widget class="QWidget" name="Kleo::Config::SMimeValidationConfigurationWidget" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>502</width>
    <height>603</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" >
     <item>
      <widget class="QCheckBox" name="intervalRefreshCB" >
       <property name="whatsThis" >
        <string>This option enables interval checking of certificate validity. You can also choose the checking interval (in hours). Note that validation is perfomed implicitly whenever significant files in ~/.gnupg change. This option therefore only affects external factors of certificate validity.</string>
       </property>
       <property name="text" >
        <string>Check certificate validity every</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="intervalRefreshSB" >
       <property name="enabled" >
        <bool>false</bool>
       </property>
       <property name="whatsThis" >
        <string>This option enables interval checking of certificate validity. You can also choose the checking interval (in hours). Note that validation is perfomed implicitly whenever significant files in ~/.gnupg change. This option therefore only affects external factors of certificate validity.</string>
       </property>
       <property name="alignment" >
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="specialValueText" >
        <string>hour</string>
       </property>
       <property name="suffix" >
        <string> hours</string>
       </property>
       <property name="minimum" >
        <number>1</number>
       </property>
       <property name="maximum" >
        <number>24</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer" >
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0" >
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QRadioButton" name="CRLRB" >
     <property name="whatsThis" >
      <string>If this option is selected, S/MIME certificates are validated using Certificate Revocation Lists (CRLs).</string>
     </property>
     <property name="text" >
      <string>Validate certificates using CRLs</string>
     </property>
     <property name="checked" >
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QRadioButton" name="OCSPRB" >
     <property name="whatsThis" >
      <string>If this option is selected, S/MIME certificates are validated online using the Online Certificates Status Protocol (OCSP). Fill in the URL of the OCSP responder below.</string>
     </property>
     <property name="text" >
      <string>Validate certificates online (OCSP)</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="OCSPGroupBox" >
     <property name="enabled" >
      <bool>false</bool>
     </property>
     <property name="title" >
      <string>Online Certificate Validation</string>
     </property>
     <layout class="QGridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="textLabel1" >
        <property name="text" >
         <string>OCSP responder URL:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="QLineEdit" name="OCSPResponderURL" >
        <property name="whatsThis" >
         <string>Enter here the address of the server for online validation of certificates (OCSP responder). The URL is usually starting with http://.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="textLabel2" >
        <property name="text" >
         <string>OCSP responder signature:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2" >
       <widget class="QCheckBox" name="ignoreServiceURLCB" >
        <property name="text" >
         <string>Ignore service URL of certificates</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="KleopatraClientCopy::Gui::CertificateRequester" native="1" name="OCSPResponderSignature" >
        <property name="whatsThis" >
         <string>Choose here the certificate with which the OCSP server signs its replies.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="doNotCheckCertPolicyCB" >
     <property name="whatsThis" >
      <string>By default, GnuPG uses the file ~/.gnupg/policies.txt to check if a certificate policy is allowed. If this option is selected, policies are not checked.</string>
     </property>
     <property name="text" >
      <string>Do not check certificate policies</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="neverConsultCB" >
     <property name="whatsThis" >
      <string>If this option is checked, Certificate Revocation Lists are never used to validate S/MIME certificates.</string>
     </property>
     <property name="text" >
      <string>Never consult a CRL</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="allowMarkTrustedCB" >
     <property name="whatsThis" >
      <string>If this option is checked while a root CA certificate is being imported, you will be asked to confirm its fingerprint and to state whether or not you consider this root certificate to be trusted. A root certificate needs to be trusted before the certificates it certified become trusted, but lightly allowing trusted root certificates into your certificate store will undermine the security of the system.</string>
     </property>
     <property name="text" >
      <string>Allow to mark root certificates as trusted</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="fetchMissingCB" >
     <property name="whatsThis" >
      <string>If this option is checked, missing issuer certificates are fetched when necessary (this applies to both validation methods, CRLs and OCSP).</string>
     </property>
     <property name="text" >
      <string>Fetch missing issuer certificates</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="requestsTabs" >
     <widget class="QWidget" name="tabHTTP" >
      <attribute name="title" >
       <string>&amp;HTTP Requests</string>
      </attribute>
      <layout class="QGridLayout" >
       <item row="0" column="0" colspan="2" >
        <widget class="QCheckBox" name="disableHTTPCB" >
         <property name="whatsThis" >
          <string>Entirely disables the use of HTTP for S/MIME.</string>
         </property>
         <property name="text" >
          <string>Do not perform any HTTP requests</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2" >
        <widget class="QCheckBox" name="ignoreHTTPDPCB" >
         <property name="whatsThis" >
          <string>When looking for the location of a CRL, the to-be-tested certificate usually contains what are known as "CRL Distribution Point" (DP) entries, which are URLs describing the way to access the CRL. The first-found DP entry is used.  With this option, all entries using the HTTP scheme are ignored when looking for a suitable DP.</string>
         </property>
         <property name="text" >
          <string>Ignore HTTP CRL distribution point of certificates</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" >
        <widget class="QRadioButton" name="honorHTTPProxyRB" >
         <property name="whatsThis" >
          <string>If this option is selected, the value of the HTTP proxy shown on the right (which comes from the environment variable http_proxy) will be used for any HTTP request.</string>
         </property>
         <property name="text" >
          <string>Use system HTTP proxy:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1" >
        <widget class="QLabel" name="systemHTTPProxy" >
         <property name="text" >
          <string>no proxy</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0" >
        <widget class="QRadioButton" name="useCustomHTTPProxyRB" >
         <property name="text" >
          <string>Use this proxy for HTTP requests: </string>
         </property>
        </widget>
       </item>
       <item row="3" column="1" >
        <widget class="QLineEdit" name="customHTTPProxy" >
         <property name="whatsThis" >
          <string>&lt;p&gt;If no system proxy is set, or you need to use a different proxy for GpgSM, you can enter its location here.&lt;/p&gt;&lt;p&gt;It will be used for all HTTP requests relating to S/MIME.&lt;/p&gt;&lt;p&gt;The syntax is host:port, for instance myproxy.nowhere.com:3128.&lt;/p&gt;</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2" >
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>320</width>
           <height>16</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabLDAP" >
      <attribute name="title" >
       <string>&amp;LDAP Requests</string>
      </attribute>
      <layout class="QGridLayout" >
       <item row="0" column="0" colspan="2" >
        <widget class="QCheckBox" name="disableLDAPCB" >
         <property name="whatsThis" >
          <string>Entirely disables the use of LDAP for S/MIME.</string>
         </property>
         <property name="text" >
          <string>Do not perform any LDAP requests</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2" >
        <widget class="QCheckBox" name="ignoreLDAPDPCB" >
         <property name="whatsThis" >
          <string>When looking for the location of a CRL, the to-be-tested certificate usually contains what are known as "CRL Distribution Point" (DP) entries, which are URLs describing the way to access the CRL. The first found DP entry is used.  With this option, all entries using the LDAP scheme are ignored when looking for a suitable DP.</string>
         </property>
         <property name="text" >
          <string>Ignore LDAP CRL distribution point of certificates</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" >
        <widget class="QLabel" name="customLDAPLabel" >
         <property name="text" >
          <string>Primary host for LDAP requests:</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1" >
        <widget class="QLineEdit" name="customLDAPProxy" >
         <property name="whatsThis" >
          <string>Entering an LDAP server here will make all LDAP requests go to that server first. More precisely, this setting overrides any specified host and port part in a LDAP URL and will also be used if host and port have been omitted from the URL. Other LDAP servers will be used only if the connection to the "proxy" failed.
The syntax is "HOST" or "HOST:PORT". If PORT is omitted, port 389 (standard LDAP port) is used.</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2" >
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>320</width>
           <height>16</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KleopatraClientCopy::Gui::CertificateRequester</class>
   <extends>QWidget</extends>
   <header>libkleopatraclient/gui/certificaterequester.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <includes>
  <include location="local" >ui/keyrequester.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>CRLRB</sender>
   <signal>toggled(bool)</signal>
   <receiver>OCSPGroupBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>29</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useCustomHTTPProxyRB</sender>
   <signal>toggled(bool)</signal>
   <receiver>customHTTPProxy</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>44</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel" >
     <x>481</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableLDAPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>ignoreLDAPDPCB</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>190</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>188</x>
     <y>471</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableLDAPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>customLDAPLabel</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>136</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>146</x>
     <y>508</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableLDAPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>customLDAPProxy</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>328</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>481</x>
     <y>508</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableHTTPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>ignoreHTTPDPCB</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>338</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>338</x>
     <y>471</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableHTTPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>honorHTTPProxyRB</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>116</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>126</x>
     <y>507</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>disableHTTPCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>useCustomHTTPProxyRB</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>92</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel" >
     <x>98</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>intervalRefreshCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>intervalRefreshSB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>282</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel" >
     <x>334</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
